### PR TITLE
add info for yaml header parse fail.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @superyyrrzz @qinezh @hellosnow @ansyral @vicancy

### CODE REVIEW

File: src/Microsoft.DocAsCode.Dfm/Rules/DfmYamlHeaderBlockRule.cs [hunk 1]
```diff
@@ -42,6 +42,7 @@ public IMarkdownToken TryMatch(IMarkdownParser parser, IMarkdownParsingContext c
             }
             catch (Exception)
             {
+                Logger.LogInfo("Invalid yaml header.", file: context.File, line: context.LineNumber.ToString());
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Dfm/Rules/DfmYamlHeaderBlockRule.cs]: also print exception message?

Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Dfm/Rules/DfmYamlHeaderBlockRule.cs]: I think no one understand the exception message.

**Comment by vicancy** [comment]
:shipit:

**Comment by hellosnow** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by qinezh** [comment]
:shipit:

**Pull request merged**